core/gui/src/app/dashboard/component/user/user-dataset/user-dataset-explorer/user-dataset-readme/user-dataset-readme.component.ts

```diff
@@ -0,0 +1,275 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges,
+  ViewChild,
+} from "@angular/core";
+import { DatasetService } from "../../../../../service/user/dataset/dataset.service";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { NotificationService } from "../../../../../../common/service/notification/notification.service";
+import { switchMap } from "rxjs/operators";
+import { of } from "rxjs";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-user-dataset-readme",
+  templateUrl: "./user-dataset-readme.component.html",
+  styleUrls: ["./user-dataset-readme.component.scss"],
+})
+export class UserDatasetReadmeComponent implements OnInit, OnChanges {
```

Comment:
In general, I want this component to be general enough. We can call it user-dataset-file-editor. It takes the file format information, and renders the corresponding editor. For now we can just support the markdown type.

Also I saw you wrote a markdown editor yourself. I am wondering if there are good libraries for this.